### PR TITLE
[WIP] Add markdown extension, update jekyll plugin documentation

### PR DESCRIPTION
**Pull request by gemfarmer**
WIP

Fixes issue(s) #1931 

[CircleCI](https://circleci.com/gh/18F/18f.gsa.gov/tree/same-headers)

[:sunglasses: PREVIEW](https://federalist.18f.gov/preview/18F/18f.gsa.gov/same-headers/)

Changes proposed in this pull request:
- Added a new default markdown, `RedcarpetExt`, which extends Redcarpet, and will allow us to add add block-level custom markdown in the future
- removes unused plugins `related_posts` and `with_pic`
- replaces markup with accessible markdown at `_posts/2016-03-18-how-user-archetypes-informed-the-draft-us-web-design-standards.md`. [View it here](https://federalist.18f.gov/preview/18F/18f.gsa.gov/same-headers/2016/03/18/how-user-archetypes-informed-the-draft-us-web-design-standards/)
  - [#concerns-and-challenges](https://federalist.18f.gov/preview/18F/18f.gsa.gov/same-headers/2016/03/18/how-user-archetypes-informed-the-draft-us-web-design-standards/#concerns-and-challenges)
  - [#concerns-and-challenges_1](https://federalist.18f.gov/preview/18F/18f.gsa.gov/same-headers/2016/03/18/how-user-archetypes-informed-the-draft-us-web-design-standards/#concerns-and-challenges_1)

/cc @mugizico

### CODE REVIEW

File: _plugins/related_posts.rb [hunk 1]
```diff
@@ -1,65 +0,0 @@
-module Jekyll
```
Comment (gemfarmer) [inline _plugins/related_posts.rb]: @gboone this doesn't appear to be used anywhere. I removed it and the related posts section still works.

[Check it out](https://federalist.18f.gov/preview/18F/18f.gsa.gov/same-headers/2016/07/29/live-streamed-demos-for-exponential-transparency-and-information-sharing/)

**Comment by gboone** [comment]
Looks like the tests are failing here because of a post that probably has a new # href now. (That means it's working, I think?)

**Comment by mugizico** [comment]
was this post added today? @gboone

**Comment by gboone** [comment]
Nope, it's [this one](18f.gsa.gov/2016/03/23/what-have-you-loved-working-on/)

**Comment by mugizico** [comment]
Looking at it!

**Comment by gemfarmer** [comment]
🎉 🎉

**Pull request merged**